SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1,9 +1,859 @@
+	TSet<UClass*> Classes = { CurrentClass };
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+
+	// WHEN
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+
+	// THEN
+	FString FileContent = LoadSchemaFileForClass(gSchemaOutputFolder, CurrentClass);
+	TestTrue("Generated Actor schema is valid", Validator.ValidateGeneratedSchemaForClass(FileContent, CurrentClass));
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_multiple_Actor_classes_WHEN_generated_schema_for_these_classes_THEN_files_with_valid_schema_exist)
+{
+	// GIVEN
+	SchemaValidator Validator;
+	TSet<UClass*> Classes =
+	{
+		ASpatialTypeActor::StaticClass(),
+		ANonSpatialTypeActor::StaticClass()
+	};
+
+	// Classes need to be sorted to have proper ids
+	Classes.Sort([](const UClass& A, const UClass& B)
+	{
+		return A.GetPathName() < B.GetPathName();
+	});
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+
+	// WHEN
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+
+	// THEN
+	bool bValidSchemaExists = true;
+	for (const auto& CurrentClass : Classes)
+	{
+		FString FileContent = LoadSchemaFileForClass(gSchemaOutputFolder, CurrentClass);
+		if(!Validator.ValidateGeneratedSchemaForClass(FileContent, CurrentClass))
+		{
+			bValidSchemaExists = false;
+			break;
+		}
+	}
+
+	TestTrue("Generated Actor schema is valid", bValidSchemaExists);
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_an_Actor_component_class_WHEN_generated_schema_for_this_class_THEN_a_file_with_valid_schema_exists)
+{
+	// GIVEN
+	SchemaValidator Validator;
+	UClass* CurrentClass = USpatialTypeActorComponent::StaticClass();
+	TSet<UClass*> Classes = { CurrentClass };
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+
+	// WHEN
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+
+	// THEN
+	FString FileContent = LoadSchemaFileForClass(gSchemaOutputFolder, CurrentClass);
+	TestTrue("Generated Actor schema is valid", Validator.ValidateGeneratedSchemaForClass(FileContent, CurrentClass));
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_an_Actor_class_with_an_actor_component_WHEN_generated_schema_for_this_class_THEN_a_file_with_valid_schema_exists)
+{
+	// GIVEN
+	SchemaValidator Validator;
+	UClass* CurrentClass = ASpatialTypeActorWithActorComponent::StaticClass();
+	TSet<UClass*> Classes = { CurrentClass };
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+
+	// WHEN
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+
+	// THEN
+	FString FileContent = LoadSchemaFileForClass(gSchemaOutputFolder, CurrentClass);
+	TestTrue("Generated Actor schema is valid", Validator.ValidateGeneratedSchemaForClass(FileContent, CurrentClass));
+
+	// CLEANUP
+	DeleteTestFolders();
```

Comment:
Perhaps we should change 'valid' to 'correct' in these tests. Just because to me 'valid' indicates it does not break the rules of schema and so could even be an empty file or the wrong schema for this class. I think we could even do other tests 'is schema valid' to test the validity first and then ones like this which is 'schema is correct for class'. Lemme know what you think.

Reply:
Changed to `expected`